### ISSUE TITLE
 invalid literal for int() with base 10: '{dbport}'

### ISSUE THREAD

**Issue by rmichaelis** [Geoportail-Luxembourg/geoportailv3#348]
After a fetch and build, we've got the following blocking error during the serve: 
 invalid literal for int() with base 10: '{dbport}'

```
Traceback (most recent call last):
  File ".build/venv/bin/pserve", line 9, in <module>
    load_entry_point('pyramid==1.6.dev0', 'console_scripts', 'pserve')()
  File "/home/mire/geoportailv3/.build/venv/src/pyramid/pyramid/scripts/pserve.py", line 58, in main
    return command.run()
  File "/home/mire/geoportailv3/.build/venv/src/pyramid/pyramid/scripts/pserve.py", line 328, in run
    global_conf=vars)
  File "/home/mire/geoportailv3/.build/venv/src/pyramid/pyramid/scripts/pserve.py", line 363, in loadapp
    return loadapp(app_spec, name=name, relative_to=relative_to, **kw)
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 247, in loadapp
    return loadobj(APP, uri, name=name, **kw)
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 272, in loadobj
    return context.create()
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 710, in create
    return self.object_type.invoke(self)
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 203, in invoke
    app = context.app_context.create()
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 710, in create
    return self.object_type.invoke(self)
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 146, in invoke
    return fix_call(context.object, context.global_conf, **context.local_conf)
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/paste/deploy/util.py", line 55, in fix_call
    val = callable(*args, **kw)
  File "/home/mire/geoportailv3/geoportailv3/__init__.py", line 23, in main
    config.include('c2cgeoportal')
  File "/home/mire/geoportailv3/.build/venv/src/pyramid/pyramid/config/__init__.py", line 788, in include
    c(configurator)
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/c2cgeoportal/__init__.py", line 406, in includeme
    'sqlalchemy.')
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/sqlalchemy/engine/__init__.py", line 384, in engine_from_config
    return create_engine(url, **options)
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/sqlalchemy/engine/__init__.py", line 362, in create_engine
    return strategy.create(*args, **kwargs)
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/sqlalchemy/engine/strategies.py", line 49, in create
    u = url.make_url(name_or_url)
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/sqlalchemy/engine/url.py", line 176, in make_url
    return _parse_rfc1738_args(name_or_url)
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/sqlalchemy/engine/url.py", line 222, in _parse_rfc1738_args
    return URL(name, **components)
  File "/home/mire/geoportailv3/.build/venv/lib/python2.7/site-packages/sqlalchemy/engine/url.py", line 60, in __init__
    self.port = int(port)
ValueError: invalid literal for int() with base 10: '{dbport}'
```

**Comment by petzlux** [comment]
Yup same here, I believe its a change in c2cgeoportal version again ... :-(

**Comment by sbrunner** [comment]
It will be fixed by: https://github.com/sbrunner/c2c.template/pull/20

**Comment by pgiraud** [comment]
Is there a workaround before the commit gets merged?

On Tue, Mar 17, 2015 at 11:27 AM, Stéphane Brunner <notifications@github.com
> wrote:

> It will be fixed by: sbrunner/c2c.template#20
> <https://github.com/sbrunner/c2c.template/pull/20>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Geoportail-Luxembourg/geoportailv3/issues/348#issuecomment-82261755>
> .
>



-- 
Pierre GIRAUD
Géomaticien, Analyste

Camptocamp France SAS
Savoie Technolac, BP 352
73377 Le Bourget du Lac, Cedex

Tel : 00 33 4 79 44 44 93
Mail : pierre.giraud@camptocamp.com
http://www.camptocamp.com

**Comment by sbrunner** [comment]
Probably by using c2c.template 1.1.0dev2

**Comment by sbrunner** [comment]
Fixed with the version 1.1.0dev4 of c2c.template
